go/worker/sentry/grpc/init.go

```diff
@@ -123,28 +153,34 @@ func New(identity *identity.Identity) (*Worker, error) {
 	if g.enabled {
 		logger.Info("Initializing gRPC sentry worker")
 
-		upstreamConn, err := initConnection(identity)
-		if err != nil {
-			return nil, fmt.Errorf("gRPC sentry worker initializing upstream connection failure: %w", err)
+		g.upstreamDialer = func(ctx context.Context) (*grpc.ClientConn, error) {
```

Comment:
Could the mutex be removed if instead the dialer just did `initConnection` and returned it (without storing it in the sentry worker struct)? 

It might be also a bit cleaner, since in that case there will always be two connections, one the proxy will use and one the policy watcher? (and each would keep track of it's own connection like both already do, and if needed, initialize more connections). I think the only difference to current impl. would be that currently the connection is ?sometimes? shared (I think currently it depends if `proxy` or `sentry` first invoke the dialer, if the connection is shared or not?).

---

On the other hand if we want to keep using one connection i think it could be done cleaner if `upstreamDialer` is it's own struct (and not part of sentry worker), that maintains a connection (or a pool of connections).

Reply:
The idea is to replace the policy watcher with a new sentry command (e.g. `UpdatePolicies` or something) that is called by the upstream node in a similar way to `SetUpstreamTLSCertificates` added in this PR.  This will allow further simplifications of the sentry code, but that should be done in a separate PR.
I'll make an issue for this after I merge the PR.